File: IvmpDotNetWrapper/PlayerManager.cpp
#include "SDK\SDK.h"

EXPORT bool GetPlayerManager_DoesExist(EntityId playerId) {
	return GetPlayerManager()->DoesExist(playerId);
}

EXPORT void GetPlayerManager_Add(EntityId playerId, char * sPlayerName) {
	GetPlayerManager()->Add(playerId, sPlayerName);
}

EXPORT bool GetPlayerManager_Remove(EntityId playerId, BYTE byteReason) {
	return GetPlayerManager()->Remove(playerId, byteReason);
}

EXPORT bool GetPlayerManager_IsNameInUse(char * szNick) {
	return GetPlayerManager()->IsNameInUse(szNick);
}

EXPORT EntityId GetPlayerManager_GetPlayerFromName(char * sNick) {
	return GetPlayerManager()->GetPlayerFromName(sNick);
}

EXPORT CPlayerInterface * GetPlayerManager_GetAt(EntityId playerId) {
	return GetPlayerManager()->GetAt(playerId);
}

File: IvmpDotNet.Core/Wrappings/PlayerManager.cs
using IvmpDotNet.SDK;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace IvmpDotNet.Wrappings {
    public class PlayerManager : IPlayerManager {
        public bool SendMessageToAll(string szMessage, int iColor, bool sqbAllowFormatting) {
            return Imports.Players.Players_SendMessageToAll(szMessage, iColor, sqbAllowFormatting);
        }

        public bool DisplayTextForAll(float fPosX, float fPosY, string szText, int iTime) {
            return Imports.Players.Players_DisplayTextForAll(fPosX, fPosY, szText, iTime);
        }

        public IPlayer GetAt(ushort playerId) {
            return new Player() {
                PlayerId = playerId
            };
        }

        public bool DoesExist(ushort playerId) {
            return Imports.GetPlayerManager.GetPlayerManager_DoesExist(playerId);
        }

        public void Add(ushort playerId, string sPlayerName) {
            Imports.GetPlayerManager.GetPlayerManager_Add(playerId, sPlayerName);
        }

        public bool Remove(ushort playerId, byte byteReason) {
            return Imports.GetPlayerManager.GetPlayerManager_Remove(playerId, byteReason);
        }

        public bool IsNameInUse(string szNick) {
            return Imports.GetPlayerManager.GetPlayerManager_IsNameInUse(szNick);
        }

        public IPlayer GetPlayerFromName(string sNick) {
            ushort id = Imports.GetPlayerManager.GetPlayerManager_GetPlayerFromName(sNick);
            return GetAt(id);
        }
    }
}
